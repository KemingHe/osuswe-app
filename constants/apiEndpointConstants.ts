// Firestore endpoints.
export const USER_PROFILE_API_ENDPOINT: string = '/api/user/profile';

// Firebase auth endpoints.
export const AUTH_SIGN_IN_API_ENDPOINT: string = '/api/auth/sign-in';
