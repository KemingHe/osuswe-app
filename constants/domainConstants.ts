export const LOCAL_DOMAIN: string = 'http://localhost:3000';
export const PRODUCTION_DOMAIN: string = 'https://osuswe.app';
