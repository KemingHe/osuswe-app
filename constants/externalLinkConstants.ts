// GitHub links.
export const GITHUB_REPO_LINK: string =
  'https://github.com/KemingHe/osuswe-app';

// Legal document links.
export const LEGAL_DISCLAIMER_LINK: string =
  'https://github.com/KemingHe/osuswe-app/blob/main/docs/legal/legal-disclaimer.md';
export const PRIVACY_POLICY_LINK: string =
  'https://github.com/KemingHe/osuswe-app/blob/main/docs/legal/privacy-policy.md';
export const TERMS_AND_CONDITIONS_LINK: string =
  'https://github.com/KemingHe/osuswe-app/blob/main/docs/legal/terms-and-conditions.md';

// OSUSWE organization links.
export const OSUSWE_ORG_HOME_LINK: string = 'https://osuswe.org';
export const OSUSWE_ORG_EVENTS_LINK: string = 'https://osuswe.org/calendar';
export const OSUSWE_ORG_MEMBERSHIP_LINK: string =
  'https://osuswe.org/how-to-become-a-member';
export const OSUSWE_ORG_OFFICERS_LINK: string =
  'https://osuswe.org/swe-officers';
export const OSUSWE_ORG_FAQ_LINK: string = 'https://osuswe.org/faq';
export const OSUSWE_ORG_EMAIL_LINK: string = 'mailto:osuswe@gmail.com';
export const OSUSWE_ORG_SLACK_LINK: string = 'https://ohiostateswe.slack.com';
export const OSUSWE_ORG_INSTAGRAM_LINK: string = 'https://instagram.com/osuswe';
export const OSUSWE_ORG_LINKEDIN_LINK: string =
  'https://il.linkedin.com/company/osuswe';

// Buckeye GPT links.
export const BUCKEYE_GPT_HOME_LINK: string = 'https://buckeye.chat';
