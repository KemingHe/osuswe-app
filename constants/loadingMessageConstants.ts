// IMPORTANT: 'Redirecting to dashboard...' is the longest string
// that can be properly displayed in the loading message component.
// DO NOT exceed this length when creating new loading messages.

// Default loading message.
export const DEFAULT_LOADING_MESSAGE: string = 'Loading...';

// Stack-Auth loading messages.
export const STACK_PROVIDER_LOADING_MESSAGE: string =
  'Securing your session...';
export const STACK_ROLE_PROVIDER_LOADING_MESSAGE: string =
  'Fetching user roles...';
export const REQUIRES_SIGNED_IN_GUARD_REDIRECTING_MESSAGE: string =
  'Redirecting to sign in...';
export const REDIRECT_ONCE_SIGNED_IN_GUARD_REDIRECTING_MESSAGE: string =
  'Redirecting to dashboard...';
export const VERIFY_EMAIL_LOADING_MESSAGE: string = 'Verifying your email...';

// Firebase loading messages.
export const FIREBASE_AUTH_INTEGRATION_LOADING_MESSAGE: string =
  'Validating your role...';
export const FIREBASE_AUTH_PROVIDER_LOADING_MESSAGE: string =
  'Signing you in...';
