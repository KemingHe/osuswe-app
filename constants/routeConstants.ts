export const HOME_ROUTE = '/';

export const AUTH_SIGN_IN_ROUTE = '/auth/sign-in';
export const AUTH_VERIFY_ROUTE = '/auth/verify';

export const USER_DASHBOARD_ROUTE = '/user/dashboard';
export const USER_PROFILE_ROUTE = '/user/profile';
