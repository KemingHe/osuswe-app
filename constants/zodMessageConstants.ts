// Error messages.
export const NONEMPTY_SELECTION_ERROR_MESSAGE: string =
  'Please make a selection';
export const UNIQUE_VALUES_ERROR_MESSAGE: string =
  'Please select each option only once';
