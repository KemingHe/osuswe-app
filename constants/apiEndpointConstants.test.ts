import * as apiEndpointConstants from '@/constants/apiEndpointConstants';

describe('(backend) api endpoint constants', () => {
  it('matches the latest snapshot', () => {
    expect(apiEndpointConstants).toMatchSnapshot();
  });
});
