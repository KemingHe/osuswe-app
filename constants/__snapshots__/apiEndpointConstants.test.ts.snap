// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`(backend) api endpoint constants > matches the latest snapshot 1`] = `
{
  "AUTH_SIGN_IN_API_ENDPOINT": "/api/auth/sign-in",
  "USER_PROFILE_API_ENDPOINT": "/api/user/profile",
}
`;
